src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.UnitConverter/UnitHandler.cs

```diff
@@ -6,108 +6,45 @@ namespace Community.PowerToys.Run.Plugin.UnitConverter
+            if (first != null)
             {
-                case Abbreviated.Both:
-                    return UnitsNet.UnitConverter.ConvertByAbbreviation(double.Parse(split[0], currentCulture), unitInfo.Name, input_first_unit, input_second_unit);
-
-                case Abbreviated.Neither:
-                    return UnitsNet.UnitConverter.ConvertByName(double.Parse(split[0], currentCulture), unitInfo.Name, input_first_unit, input_second_unit);
-
-                case Abbreviated.Second:
-                    UnitInfo first = Array.Find(unitInfo.UnitInfos, info => info.Name.ToLower() == input_first_unit.ToLower());
-                    UnitParser.Default.TryParse(split[3], unitInfo.UnitType, out Enum second_unit);
-                    return UnitsNet.UnitConverter.Convert(double.Parse(split[0], currentCulture), first.Value, second_unit);
-
-                case Abbreviated.First:
-                    UnitParser.Default.TryParse(split[1], unitInfo.UnitType, out Enum first_unit);
-                    UnitInfo second = Array.Find(unitInfo.UnitInfos, info => info.Name.ToLower() == input_second_unit.ToLower());
-                    return UnitsNet.UnitConverter.Convert(double.Parse(split[0], currentCulture), first_unit, second.Value);
+                return first.Value;
+            }
 
-                case Abbreviated.NotFound:
-                default:
-                    return double.NaN;
+            if (UnitParser.Default.TryParse(unit, unitInfo.UnitType, out Enum enum_unit))
+            {
+                return enum_unit;
             }
+
+            return null;
         }
 
         /// <summary>
-        /// Given a split array of user input, parses the input for abbreviations (e.g. "1 cm in meters").
+        /// Given user unit input, converts it. (E.g "1 foot in cm").
         /// </summary>
-        /// <returns>A tuple consisting of an Abbreviated enum and QuantityInfo.</returns>
-        public static (Abbreviated Abbreviated, QuantityInfo UnitInfo) ParseInputForAbbreviation(string[] split, QuantityType quantityType)
+        /// <returns>The converted value as a double.</returns>
+        public static double ConvertInput(string[] split, QuantityType quantityType, CultureInfo currentCulture)
         {
-            string input_first_unit = split[1];
-            string input_second_unit = split[3];
-
-            QuantityInfo unit_info = Quantity.GetInfo(quantityType);
-            bool first_unit_is_abbreviated = UnitParser.Default.TryParse(split[1], unit_info.UnitType, out Enum _);
-            bool second_unit_is_abbreviated = UnitParser.Default.TryParse(split[3], unit_info.UnitType, out Enum _);
-
-            // 3 types of matches:
-            // a) 10 ft in cm (double abbreviation)
-            // b) 10 feet in centimeter (double unabbreviated)
-            // c) 10 feet in cm (single abbreviation)
+            string firstUnitString = split[1];
```

Comment:
Is it guaranteed that the `split` array has and would always have exact 4 values? IMO at first, the array size should be checked to avoid out of range exceptions.

Reply:
Agree, but let's do it in a separate pr. I plan to replace the split array here with class.